xilem/src/view/memoize.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2024 the Xilem Authors
+// SPDX-License-Identifier: Apache-2.0
+
+use std::any::Any;
+
+use masonry::{widget::WidgetMut, WidgetPod};
+
+use crate::{MasonryView, MessageResult, ViewCx, ViewId};
+
+pub struct Memoize<D, F> {
+    data: D,
+    child_cb: F,
+}
+
+pub struct MemoizeState<T, A, V: MasonryView<T, A>> {
+    view: V,
+    view_state: V::ViewState,
+    dirty: bool,
+}
+
+impl<D, V, F> Memoize<D, F>
+where
+    F: Fn(&D) -> V,
+{
+    const ASSERT_CONTEXTLESS_FN: () = {
+        assert!(
+            std::mem::size_of::<F>() == 0,
+            "The callback is not allowed to be a function pointer or a closure capturing context"
```

Comment:
We should explain why here

Reply:
Is the content of the assert in the last commit better?